subsystem/infra_env_test.go

```diff
@@ -46,6 +48,65 @@ var internalRegisterInfraEnv = func(clusterID *strfmt.UUID, imageType models.Ima
 	return request.GetPayload()
 }
 
+func validateKernelArgs(infraEnv *models.InfraEnv, expectedKargs models.KernelArguments) {
+	if expectedKargs == nil {
+		Expect(infraEnv.KernelArguments).To(BeNil())
+	} else {
+		Expect(infraEnv.KernelArguments).ToNot(BeNil())
+		var kargs models.KernelArguments
+		Expect(json.Unmarshal([]byte(*infraEnv.KernelArguments), &kargs)).ToNot(HaveOccurred())
+		Expect(kargs).To(Equal(expectedKargs))
+	}
+}
+
+var _ = Describe("Register InfraEnv- kernel arguments", func() {
+	register := func(kernelArgs models.KernelArguments) (*installer.RegisterInfraEnvCreated, error) {
+		return userBMClient.Installer.RegisterInfraEnv(context.Background(), &installer.RegisterInfraEnvParams{
+			InfraenvCreateParams: &models.InfraEnvCreateParams{
+				Name:             swag.String("test-infra-env"),
+				OpenshiftVersion: openshiftVersion,
+				PullSecret:       swag.String(pullSecret),
+				SSHAuthorizedKey: swag.String(sshPublicKey),
+				ImageType:        models.ImageTypeMinimalIso,
+				KernelArguments:  kernelArgs,
+			},
+		})
+	}
+
+	It("happy flow", func() {
+		kargs := models.KernelArguments{
+			{
+				Operation: models.KernelArgumentOperationAppend,
+				Value:     "p1",
+			},
+			{
+				Operation: models.KernelArgumentOperationAppend,
+				Value:     `p2="this is an argument"`,
+			},
+		}
+		res, err := register(kargs)
+		Expect(err).ToNot(HaveOccurred())
+		validateKernelArgs(res.Payload, kargs)
+	})
+
+	DescribeTable("unsupported cases",
+		func(operation, value string) {
+			kargs := models.KernelArguments{
+				{
+					Operation: operation,
+					Value:     value,
+				},
+			}
+			_, err := register(kargs)
+			Expect(err).To(HaveOccurred())
+		},
+		Entry("operation is replace", models.KernelArgumentOperationReplace, "p1"),
```

Comment:
from the name it's not clear what is not supported in this case? 
is it the operator or the argument? 
same question for the next entry

Reply:
maybe `unsupported replace operation` it wasn't clear because i was thinking that replace of p1 is invalid because it doesn't exist in the db

Reply:
OK